### PR TITLE
Removing old DockArea

### LINKED ISSUE #177: BECDock drag and drop, stack on each other instead of tab
Docks should be always tabbed when dropped on other dock. However the first dropped dock is sometimes putted on the top:

dock_tab
 Migrated from GitLab issue https://gitlab.psi.ch/bec/bec_widgets/-/issues/173

### LINKED ISSUE #477: BECConnector remove parent_dock from kwargs and adjust logic to parent-child
parent_dock kwarg is not relevant anymore with new qt hierarchy approach, we do not have to hardcode it anymore for dock areas.
 Migrated from GitLab issue https://gitlab.psi.ch/bec/bec_widgets/-/issues/473

### LINKED ISSUE #606: DockArea: Dock.detach() removes itself from DockArea RPC nesting, even if then it is manually putted back

### LINKED ISSUE #738: [BUG]: DockArea becomes unresponsive
### Provide a brief description of the bug.

When using the dock area, there seems to be a sequence of actions that can put the GUI into an unresponsive state. Even closing the widget doesn't help as the launcher's DockArea button also doesn't work anymore.

### Describe what you expected to happen and what actually happened.

_No response_

### Outline the steps that lead to the bug's occurrence. Be specific and provide a clear sequence of actions.

* Open the launcher / start the dock area
* Add a positioner box
* Add a scan control widget
* Remove the positioner box
* Remove the scan control widget

The DockArea is now unresponsive. 

### bec_widgets version

main

### bec core version

main

### Any extra info / data? e.g. log output...

I added a logger to the create widget slot (i.e. the slot triggered by clicking the action in the dock area's toolbar). It doesn't receive a call. 

### Related issues

None

### LINKED ISSUE #956: Remove PyQtGraph dock area from the project and replace by ADS one
For V3 breaking changes

### PR DESCRIPTION
**Pull request by wyzula-jan**
## Description

Removes old `BECDockArea` and replace all occurrences with new `AdvancedDockArea` based on ADS. During testing I also discovered few bugs related to `AdvancedDockArea` which I fixed (see Type of changes lower).

## Related Issues

- closes #956 
- closes #738 
- closes #606
- closes #477 
- closes #177 

### **BECMainWindow**
- Delete on exit added, main window was leaking and was not deleted.

### **AdvancedDockArea**:
- **bug** `delete_all()` deletes also floating widgets
- **bug** `load_profile` → changing profile was just hiding widgets, not deleting them making them leak forever...
- **UX** → newly created profiles in gui are quick select by default
- **bug** → when creating new namespace for profiles for ADS, the ADS will ensure that at least one profile `general` will exist
- **bug** in ADS the dock_list and dock_map do not make sense for CLI and was leading to timeout due to serialization (CDockWidgets are not part of the cli client) → removed from CLI API completely 
- **UX** → cli companion window and main app dock_area shares the same namespace for profiles

### **BECDockArea and LaunchWindow**
- old BECDockArea based on pyqtgraph completely removed from BEC Widgets and replaced everywhere with ADS
- Launcher adapted UX, able to launch ADS with selected profile, geometry of launched widgets fixed

## How to test

- open CLI and interact with the new dock area in similar way as with the old one

## Potential side effects

[Describe any potential side effects or risks of merging this PR.]

## Screenshots / GIFs (if applicable)

[Include any relevant screenshots or GIFs to showcase the changes made.]

## Additional Comments

One test in `test_user_interaction_e2e.py` is disabled, passes locally but I can't get it work on CI.

### CODE REVIEW

File: bec_widgets/widgets/containers/advanced_dock_area/advanced_dock_area.py [hunk 1]
```diff
@@ -179,6 +181,44 @@ def __init__(
...
+
+        Returns:
+            bool: True if a profile was created, False otherwise.
+        """
+        namespace = self.profile_namespace
+        try:
+            existing_profiles = list_profiles(namespace)
+        except Exception as exc:  # pragma: no cover - defensive guard
+            logger.warning(f"Unable to enumerate profiles for namespace '{namespace}': {exc}")
+            return False
```
Comment (cappel89) [inline bec_widgets/widgets/containers/advanced_dock_area/advanced_dock_area.py]: Doesn't this mean that there is no profile available? So you would have to create on in that case?

File: tests/end-2-end/test_bec_dock_rpc_e2e.py [hunk 2]
```diff
@@ -19,56 +19,33 @@ def check_dock_area_registered():
     qtbot.waitUntil(check_dock_area_registered, timeout=5000)
     assert hasattr(gui, "cool_dock_area")
 
-    dock = dock_area.new("dock_0")
+    widget = dock_area.new("Waveform", object_name="cool_waveform")
```
Comment (cappel89) [inline tests/end-2-end/test_bec_dock_rpc_e2e.py]: This used to be called name, now the user would set `object_name` directly? 

In addition, by removing 'name' from `new()`of the the new dock, we are missing important checks. The old `BECDock` was checking that if you set a 'name' that it is a valid name (-> dock.py - 306 `WidgetContainerUtils.raise_for_invalid_name(name, container=self)`). 

Now, I can choose invalid Python names and break the namespace..

**Review by cappel89 — COMMENTED** [review]
I mostly checked code and test changes so far. I will play around with the new dock_area a bit more. Looks nice, overall larger than the previous one, which I personally prefer. There are a few things in the RPC interface that surprised me, please check the comments for that

File: bec_widgets/widgets/containers/advanced_dock_area/advanced_dock_area.py (hunk 1 above)
Comment (wyzula-jan) [reply bec_widgets/widgets/containers/advanced_dock_area/advanced_dock_area.py]: By default there should be always at least one profile which is called "general", this ensures it (for example for the first start after deployment).

File: tests/end-2-end/user_interaction/test_user_interaction_e2e.py [hunk 3]
```diff
@@ -555,41 +536,40 @@ def test_widgets_e2e_positioner_control_line(
...
-
-    # Test rpc calls
-    dev = bec.device_manager.devices
-    scans = bec.scans
-    # Do a scan
-    scans.line_scan(dev.samx, -3, 3, steps=50, exp_time=0.01, relative=False).wait()
-
-    # Test removing the widget, or leaving it open for the next test
-    maybe_remove_dock_area(qtbot, gui=gui, random_int_gen=random_generator_from_seed)
+# TODO passes locally, fails on CI for some reason...
```
[the remark this answers is not in the document]
Comment (wyzula-jan) [reply tests/end-2-end/user_interaction/test_user_interaction_e2e.py]: issue created

File: bec_widgets/cli/server.py [hunk 4]
```diff
@@ -107,17 +107,19 @@ def _run(self):
...
-        self.launcher_window = LaunchWindow(gui_id=f"{self.gui_id}:launcher")
+        if self.gui_class:
+            self.launcher_window = LaunchWindow(
+                gui_id=f"{self.gui_id}:launcher",
+                launch_gui_class=self.gui_class,
+                launch_gui_id=self.gui_class_id,
+            )
+        else:
+            self.launcher_window = LaunchWindow(gui_id=f"{self.gui_id}:launcher")
         self.launcher_window.setAttribute(Qt.WA_ShowWithoutActivating)  # type: ignore
```
[the remark this answers is not in the document]
Comment (wyzula-jan) [reply bec_widgets/cli/server.py]: I don't understand the window is defined already with or without gui_class.

File: tests/end-2-end/test_bec_dock_rpc_e2e.py (hunk 2 above)
Comment (wyzula-jan) [reply tests/end-2-end/test_bec_dock_rpc_e2e.py]: Implemented on the level of becconnector so all object_names will be always sanitized

Comment (cappel89) [reply tests/end-2-end/test_bec_dock_rpc_e2e.py]: I think the current fix is not sufficient. You can still type in names that are invalid. I am also uncertain whether we should really try to be smart and sanitize names to this degree. Maybe it is okay to remove trailing whitespaces. We can discuss to replace other whitespaces with '_', but as seen in the screenshot below, there are still some names that pass the method and create invalid Python namespace.

Please check `WidgetContainerUtils.raise_for_invalid_name(name, container=self)` which was used before.

Screenshot 2026-01-13 at 22 06 53

Additionally, you can see below that there is a namespace that needs to be protected, e.g. 'new', basically methods from the dock_area that are used.
Please check the WidgetContainerUtils (https://github.com/bec-project/bec_widgets/blob/fix/remove-old-dock-area/bec_widgets/utils/container_utils.py#L10) for implementation of valid `object_names`. They are basically exactly for that.

Screenshot 2026-01-13 at 22 13 02

**Review by cappel89 — COMMENTED** [review]
I am not sure if I understand how I would work with profiles currently. What exactly should be stored when, to me it looks as if using multiple dock_area has very weird side effects. I will attach some videos below. Profiles seem to be saved per default whenever you switch profiles within a dock_area or if you exit a dock_area. If you delete widgets from a profile, this also gets saved.  
This is very confusing in particular if multiple dock_areas are open (or the main_app). Having multiple dock_areas open can result in a state where the profile, e.g. 'general' seems to be out of sync, and the state from 2 different dock_areas seems to compete in storing the state. This can be seen if a new dock_area is create which then either has then a layout with widgets based on who stored the profile last (e.g. through exiting the dock_area, or switching the profile).

 I don't think this behavior should be the user experience we want for them.

File: tests/end-2-end/test_bec_dock_rpc_e2e.py (hunk 2 above)
Comment (cappel89) [reply tests/end-2-end/test_bec_dock_rpc_e2e.py]: We've done something in this direction already here. Looking at it, this should possibly be usable right away..

WidgetContainerUtils (https://github.com/bec-project/bec_widgets/blob/fix/remove-old-dock-area/bec_widgets/utils/container_utils.py#L10)

Comment (wakonig) [reply tests/end-2-end/test_bec_dock_rpc_e2e.py]: Depends a bit if you want to sanitize the input or raise for invalid names. The method you referenced raises for invalid names. My previous suggestion would sanitize. Either way is fine with me but it might be good to move it to bec lib to share across repos

Comment (cappel89) [reply tests/end-2-end/test_bec_dock_rpc_e2e.py]: It's True that it raises, but that is easy to adjust.  Raise or not raise, I also have no preference. What I mean above was that I believe that`WidgetContainerUtils` have methods to perform all relevant checks needed. That's why I referenced them in the very initial comment. This does not oppose your suggestion to use `slugify` to sanitize names, but I think you still need sufficient checks if this name is allowed. Those should probably live in bec_lib in the long run, I also agree with that.

File: tests/end-2-end/test_rpc_widgets_e2e.py [hunk 5]
```diff
@@ -129,19 +122,17 @@ def test_available_widgets(qtbot, connected_client_gui_obj):
         #############################
 
         # Now we remove the widget again
-        dock_name = dock.object_name
-        dock_id = dock._gui_id
         widget_id = widget._gui_id
-        dock_area.delete(dock.object_name)
+        widget.remove()
```
[the remark this answers is not in the document]
Comment (wyzula-jan) [reply tests/end-2-end/test_rpc_widgets_e2e.py]: changed to `delete` on the level of dock_area

**Review by cappel89 — APPROVED** [review]
LGTM, but the coverage is quite low. Could you make an issue to improve coverage for `name_utils` and `client_utils`.


Image
